pages/agent/v3/signed_pipelines.md

```diff
@@ -0,0 +1,162 @@
+
+Note that the value of `--alg` must be a valid [JSON Web Signing Algorithm](https://datatracker.ietf.org/doc/html/rfc7518#section-3). The agent does not support the `none` algorithm, or any signature algorithms in the `RS` series (any signing algorithms that use RSASSA-PKCS1 v1.5 signatures, `RS256` for example).
+
+<details>
+  <summary>Why doesn't the agent support RSASSA-PKCS1 v1.5 signatures?</summary>
+  <p>In short, RSASSA-PKCS1 v1.5 signatures are less secure than the newer RSA-PSS signatures. While RSASSA-PKCS1 v1.5 signatures are still relatively secure, we want to encourage our users to use the most secure algorithms possible, so when using RSA keys, we only support RSA-PSS signatures. We also recommend looking into ECDSA and EdDSA signatures, which are more secure than RSA signatures.</p>
+</details>
+
+#### Algorithm options
+
+Signed pipelines support many different signing algorithms, which you can simplify into two categories:
+
+- **Symmetric**—where signing and verification use the same keys.
+- **Asymmetric**—where signing and verification use different keys.
+
+We recommend using an asymmetric signing algorithm, as it creates a permissions boundary between the agents that upload and sign pipelines, and agents that run other jobs and can verify signatures. This means that if an attacker compromises an agent with verification keys, they can't modify the steps without detection.
+
+When using asymmetric signing algorithms, we recommend having multiple disjoint pools of agents, each using a different [queue](/docs/agent/v3/queues). One pool should be the _uploaders_ and have access to the private keys. Another pool should be the _runners_ and have access to the public keys. This creates a security boundary between the agents that upload and sign pipelines and the agents that run jobs and verify signatures.
+
+Regarding your specific algorithm choice, any of the supported asymmetric signing algorithms are fine and will be secure. If you're not sure which one to use, `EdDSA` is proven to be secure, has a modern design, wasn't designed by a Nation State Actor, and produces nice short signatures.
+
+For more information on what algorithms the agent supports, run:
+
+```sh
+buildkite-agent tool keygen --help
+```
+
+### 2. Configure the agents
+
+Next, you need to configure your agents to use the keys you generated. On agents that upload pipelines, add the following to the agent's config file:
+
+```ini
+job-signing-jwks-path=<path to private keyset>
+job-signing-key-id=<the key id you generated earlier>
+job-verification-jwks-path=<path to public keyset>
+```
+
+This ensures that whenever those agents upload steps to Buildkite, they'll generate signatures using the private key you generated earlier. It also ensures that those agents verify the signatures of any steps they run, using the public key.
+
+On instances that verify jobs, add:
+
+```ini
+job-verification-jwks-path=<path to verification keys>
+```
+
+### 3. Sign all steps
```

Comment:
I tried to avoid introducing new terms with "configured" and "uploaded" steps. Hopefully this works